cast/tests/utils/prepare_for_tests.sh

```diff
@@ -5,20 +5,43 @@ if ! which starknet-devnet > /dev/null 2>&1; then
   exit 1
 fi
 
-COMPILER_DIRECTORY="$(git rev-parse --show-toplevel)/cast/tests/utils/"
+DIRECTORY="$(git rev-parse --show-toplevel)/cast/tests/utils/"
 CAIRO_REPO="https://github.com/starkware-libs/cairo/releases/download"
-COMPILER_VERSION="v1.0.0"
+COMPILER_VERSION="v1.1.1"
+
+SCARB_REPO="https://github.com/software-mansion/scarb/releases/download"
+SCARB_VERSION="v0.4.1"
 
 if [ ! -x "$COMPILER_DIRECTORY/cairo/bin/starknet-sierra-compile" ]; then
   if [[ $(uname -s) == 'Darwin' ]]; then
-    wget "$CAIRO_REPO/$COMPILER_VERSION/release-aarch64-apple-darwin.tar" -P "$COMPILER_DIRECTORY" || exit 1
-    pushd "$COMPILER_DIRECTORY"
-    tar -xvf "$COMPILER_DIRECTORY/release-aarch64-apple-darwin.tar" cairo/bin/starknet-sierra-compile || exit 1
+    wget "$CAIRO_REPO/$COMPILER_VERSION/release-aarch64-apple-darwin.tar" -P "$DIRECTORY" || exit 1
+    pushd "$DIRECTORY"
+    tar -xvf "$DIRECTORY/release-aarch64-apple-darwin.tar" cairo/bin/starknet-sierra-compile || exit 1
+    popd
+
+  elif [[ $(uname -s) == 'Linux' ]]; then
+    wget "$CAIRO_REPO/$COMPILER_VERSION/release-x86_64-unknown-linux-musl.tar.gz" -P "$DIRECTORY" || exit 1
+    pushd "$DIRECTORY"
+    tar -xzvf "$DIRECTORY/release-x86_64-unknown-linux-musl.tar.gz" cairo/bin/starknet-sierra-compile || exit 1
+    popd
+  fi
+fi
+
+if [ ! -x "$COMPILER_DIRECTORY/scarb*/bin/scarb" ]; then
```

Comment:
1. can scarb do multiple versions? if so, we could just check if there's required scarb version in system
2. if not, let's get move `if` statement to separate function and reuse it